File: src/11.binarySearch/07.275.go
package array_greedy

func HIndex(citations []int) int {
	n := len(citations)
	l := 1
	r := n

	for l <= r {
		mid := l + (r-l)/2

		if citations[n-mid] >= mid {
			l = mid + 1
		} else {
			r = mid - 1
		}
	}

	return r
}

File: src/11.binarySearch/07.275.ts
export function hIndex(citations: number[]): number {
    const n = citations.length;
    let l = 1;
    let r = n;

    while (l <= r) {
        const mid = l + Math.floor((r - l) / 2);

        if (citations[n - mid] >= mid) {
            l = mid + 1;
        } else {
            r = mid - 1;
        }
    }
    return r;
}
